src/App.tsx

```diff
@@ -1,44 +1,155 @@
-import { useState } from "react";
-import reactLogo from "./assets/react.svg";
-import viteLogo from "/vite.svg";
+import { useEffect, useState } from "react";
 import "./App.css";
+import { queryFirebase, getLocationDict } from "./firebase";
+import {
+    SUBMIT_PACKING,
+    packingStatusUrl,
+    getLogsUrl,
+} from "./constants/apiEndpoints";
+import {
+    AWSBatchJobsResponse,
+    CloudWatchLogsResponse,
+    StringDict,
+} from "./types";
 
 function App() {
-    const [count, setCount] = useState(0);
-    const [text, setText] = useState("");
+    const [recipes, setRecipes] = useState<StringDict>({});
+    const [configs, setConfigs] = useState<StringDict>({});
+    const [selectedRecipe, setSelectedRecipe] = useState("");
+    const [selectedConfig, setSelectedConfig] = useState("");
+    const [jobId, setJobId] = useState("");
+    const [jobStatus, setJobStatus] = useState("");
+    const [logStreamName, setLogStreamName] = useState("");
+    const [jobLogs, setJobLogs] = useState<string[]>([]);
+
+    const submitRecipe = async () => {
+        let url = `${SUBMIT_PACKING}?recipe=${selectedRecipe}`;
```

Comment:
nit: maybe we should move this into `constants/apiEndpoints`, following the same pattern as `packingStatusUrl` and `getLogsUrl`?